hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/ContractBurnHTSV2SecurityModelSuite.java

```diff
@@ -0,0 +1,693 @@
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        // The signer and the token have a threshold key with the signer's public key
+                        // and the wrong contract id (MINT_CONTRACT)
+                        contractCall(
+                                HTS_CALLS,
+                                "burnTokenCall",
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                BigInteger.valueOf(amountToBurn),
+                                new long[0])
+                                .via(SIGNER_MINTS_WITH_SIGNER_PUBLIC_KEY_AND_WRONG_CONTRACT_ID)
+                                .gas(GAS_TO_OFFER)
+                                .alsoSigningWithFullPrefix(SIGNER)
+                                .payingWith(SIGNER),
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
+                                .andAllChildRecords()
+                                .logged(),
+                        // Create a key with thresh 1/2 with sigs: new ed25519 key, contractId of HTS_CALLS contract
+                        // Here the key has the contract`id of the correct contract
+                        newKeyNamed(THRESHOLD_KEY).shape(TRESHOLD_KEY_SHAPE.signedWith(sigs(ON, HTS_CALLS))),
+                        // Set the token's supply key to the initial one
+                        tokenUpdate(FUNGIBLE_TOKEN).supplyKey(TOKEN_TREASURY),
+                        // Update the Signer with the correct threshold key
+                        cryptoUpdate(SIGNER).key(THRESHOLD_KEY),
+                        // Test Case 3: Signer paying and signing a token burn transaction, when the token
+                        // is expected to  be burned by the token treasury account
+                        // SIGNER → call → CONTRACT → call → PRECOMPILE
+                        // The token has no updated supply key. The signer has the correct threshold key
+                        contractCall(
+                                HTS_CALLS,
+                                "burnTokenCall",
+                                HapiParserUtil.asHeadlongAddress(
+                                        asAddress(spec.registry().getTokenID(FUNGIBLE_TOKEN))),
+                                BigInteger.valueOf(amountToBurn),
+                                new long[0])
+                                .via(TOKEN_HAS_NO_UPDATED_KEY)
+                                .gas(GAS_TO_OFFER)
+                                .signedBy(SIGNER)
+                                .payingWith(SIGNER),
+                        getTokenInfo(FUNGIBLE_TOKEN).hasTotalSupply(initialAmount),
+                        getTxnRecord(SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS)
```

Comment:
The name of the call is wrong here, should be "TOKEN_HAS_NO_UPDATED_KEY" instead of "SIGNER_AND_TOKEN_HAVE_NO_UPDATED_KEYS".

Reply:
fixed